EIPS/eip-7942.md

```diff
@@ -0,0 +1,191 @@
+
+Given the desired value $p$, the value of $\vartheta$ can be calculated using $\vartheta = \lfloor\mu + \sigma \cdot \Phi^{-1}(1 - p)\rfloor,$ where $\mu = f'$, $\sigma^2 = f'(1 - f/n)$, and $\Phi^{-1}$ is the inverse of the cumulative distribution function of the normal distribution. We show some concrete examples of the ratio of Byzantine validators in a committee.
+
+|  n     \ p   | 10⁻⁶  | 10⁻⁷  | 10⁻⁸  | 10⁻⁹  |
+|-------------|--------|--------|--------|--------|
+| 2¹⁴         | 0.4316 | 0.4414 | 0.4492 | 0.4570 |
+| 2¹⁶         | 0.3828 | 0.3872 | 0.3916 | 0.3955 |
+| 2¹⁸         | 0.3580 | 0.3603 | 0.3625 | 0.3645 |
+| 2²⁰         | 0.3457 | 0.3468 | 0.3479 | 0.3489 |
+
+Ethereum now has approximately $2^{20}$ validators. The maximum desirable probability $10^{-6}$ means that the protocol fails once every $10^{6}$ slots, i.e., 138 days. If the desirable failure probability is $10^{-9}$, malicious reorganization occurs once every 380 years. We can then draw the following conclusion: 
+
+**For a large $n$, the ratio of Byzantine validators is closer to $f/n$. In this way, we can simply set $\vartheta$ as one-third of the committee size.**
+
+### Protocol Changes
+
+We make three main changes:
+
+1. **Modify the selection rule of parent.** As mentioned in the overview, we require that a block $b$ in slot $t$ must include $\vartheta+1$ attestations to prove that the parent of block $b$ is a valid output in slot $t-1$. Therefore, the parent of block $b$ is the $AA_{t-1}$. Such a block is a stable block. If no $AA_{t-1}$ exists, the parent of block $b$ is the output of the fork choice rule. Such a block is an unstable block.
+
+2. **Replace HLMD-GHOST with Longest Stable Chain fork choice rule.** We replace the HLMD GHOST with the longest chain as the fork choice rule. Our longest chain fork choice rule is very simple: it outputs the head of the chain with the most stable blocks. In case of a tie, the longest chain fork choice rule chooses the chain such that the leaf block has the largest slot number.
+
+3. **Blocks carry a reference `u` to recent unstable blocks to preserve transaction continuity.** We additionally include a field `u` in a block b. Each proposer sets its `u` field as the hash of an unstable block. If there is no such unstable block, set `u` as None. The transactions in the unstable blocks should not conflict with the transactions in the longest chain. The transactions in `u` can also be finalized once block b (that includes `u`) is finalized.
+
+4. **Simplify the filtering rule and the justification process.** We simplify the filtering rule as it only considers the chain that justifies the last justified checkpoint. Additionally, the justification process only updates when a chain crosses the epoch boundary.
+
+### Pseudocode
+
+The modifications are based on validator.md and fork-choice.md.
```

Comment:
could you please remove .md from here and just say : are based on validator and fork-choice specs from <spec url>